File: wxcloudrun/models/policy.py
from sqlalchemy import Column, Integer, String, Text, TIMESTAMP, Enum, UniqueConstraint
from sqlalchemy.sql import func
from wxcloudrun.core.database import Base


class Policy(Base):
    __tablename__ = 'policies'

    id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    type = Column(Enum('terms', 'privacy', name='policy_type_enum'), nullable=False, index=True)
    version = Column(String(32), nullable=False)
    title = Column(String(200), nullable=True)
    content = Column(Text, nullable=False)
    format = Column(Enum('markdown', 'html', 'text', name='policy_format_enum'), nullable=False, default='markdown')
    locale = Column(String(16), nullable=False, default='zh-CN', index=True)
    status = Column(Enum('draft', 'published', 'archived', name='policy_status_enum'), nullable=False, default='draft', index=True)
    effective_at = Column(TIMESTAMP, nullable=True)
    created_at = Column(TIMESTAMP, nullable=False, server_default=func.now())
    updated_at = Column(TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now())

    __table_args__ = (
        UniqueConstraint('type', 'version', 'locale', name='uq_policy_type_version_locale'),
    )

File: src/main/java/com/babykeke/backend/entity/Policy.java
package com.babykeke.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * 政策协议实体
 */
@Data
@Entity
@Table(name = "policies",
    uniqueConstraints = {
        @UniqueConstraint(name = "uq_policy_type_version_locale", columnNames = {"type", "version", "locale"})
    },
    indexes = {
        @Index(name = "idx_type", columnList = "type"),
        @Index(name = "idx_locale", columnList = "locale"),
        @Index(name = "idx_status", columnList = "status")
    }
)
public class Policy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false, columnDefinition = "ENUM('terms', 'privacy')")
    private PolicyType type;

    @Column(name = "version", nullable = false, length = 32, columnDefinition = "VARCHAR(32)")
    private String version;

    @Column(name = "title", length = 200, columnDefinition = "VARCHAR(200)")
    private String title;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(name = "format", nullable = false, columnDefinition = "ENUM('markdown', 'html', 'text') DEFAULT 'markdown'")
    private PolicyFormat format = PolicyFormat.markdown;

    @Column(name = "locale", nullable = false, length = 16, columnDefinition = "VARCHAR(16) DEFAULT 'zh-CN'")
    private String locale = "zh-CN";

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, columnDefinition = "ENUM('draft', 'published', 'archived') DEFAULT 'draft'")
    private PolicyStatus status = PolicyStatus.draft;

    @Column(name = "effective_at", columnDefinition = "TIMESTAMP")
    private LocalDateTime effectiveAt;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false, columnDefinition = "TIMESTAMP")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false, columnDefinition = "TIMESTAMP")
    private LocalDateTime updatedAt;
}
